dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3DeleteBucketTask.java

```diff
@@ -0,0 +1,50 @@
+package com.spectralogic.dsbrowser.gui.services.tasks;
+
+import com.spectralogic.ds3client.Ds3Client;
+import com.spectralogic.ds3client.commands.spectrads3.DeleteBucketSpectraS3Request;
+import com.spectralogic.ds3client.networking.FailedRequestException;
+import com.spectralogic.dsbrowser.gui.util.Ds3Task;
+import javafx.concurrent.WorkerStateEvent;
+import javafx.event.Event;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Ds3DeleteBucketTask extends Ds3Task {
+    private final static Logger LOG = LoggerFactory.getLogger(Ds3DeleteBucketTask.class);
+
+    private final Ds3Client ds3Client;
+    private final String bucketName;
+    private String errorMsg;
+
+    public Ds3DeleteBucketTask(final Ds3Client ds3Client, final String bucketName) {
+        this.ds3Client = ds3Client;
+        this.bucketName = bucketName;
+    }
+
+    @Override
+    protected String call() throws Exception {
+        try {
+            ds3Client.deleteBucketSpectraS3(new DeleteBucketSpectraS3Request(bucketName).withForce(true));
+            return "Success";
+        } catch (final FailedRequestException fre) {
```

Comment:
If there are no differences with the Exception block bellow this then remove this block